### PR TITLE
Restructure

### PR DESCRIPTION
**Pull request by DanielElisenberg**
# RESTRUCTURE

Structured repo to be more similar to our other repos.

### Whats's changed?
* Source and tests structured into directories steps, schema, repository, exceptions, components
* Removed utils
* Added new tests
* Moved some functions from steps that was more appropriate for local_storage
* Some general code clean up

### What's next?
* Implementing injection of centrally defined identifier variables

### CODE REVIEW

**Comment by DanielElisenberg** [comment]
Working on the 3.(7|8) problem 🏃🏻

File: microdata_validator/schema/__init__.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def inline_metadata_references(metadata_file_path: Path,
+                               metadata_ref_directory: Path) -> dict:
+    def recursive_ref_insert(metadata_object: dict):
+        for key, value in metadata_object.items():
+            if isinstance(value, dict):
+                if "$ref" in value:
+                    metadata_object[key] = local_storage.load_json(
+                        metadata_ref_directory / Path(value["$ref"])
```
Comment (pawbu) [inline microdata_validator/schema/__init__.py:33]: Nice syntax

File: tests/unit/repository/test_local_storage.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+from microdata_validator.repository import local_storage
+
+RESOURCE_DIR = Path('tests/resources/repository')
+UNSORTED_DATASET_PATH = RESOURCE_DIR / 'unsorted_dataset.csv'
+SORTED_DATASET_PATH = RESOURCE_DIR / 'sorted_dataset.csv'
+SQLITE_FILE_PATH = RESOURCE_DIR / 'sql_dataset.db'
+
+
+def test_sql_read_sorted():
```
Comment (pawbu) [inline tests/unit/repository/test_local_storage.py:13]: 💯

**Review by pawbu — APPROVED** [review]
Looks good

**Pull request merged**